functional-tests/src/test/groovy/io/micronaut/gradle/shadow/ShadowJarSpec.groovy

```diff
@@ -0,0 +1,80 @@
+package io.micronaut.gradle.shadow
+
+import io.micronaut.gradle.fixtures.AbstractFunctionalTest
+import org.gradle.testkit.runner.TaskOutcome
+import spock.lang.Issue
+
+class ShadowJarSpec extends AbstractFunctionalTest {
+
+    @Issue("https://github.com/micronaut-projects/micronaut-gradle-plugin/issues/406")
+    def "merges service files when building shadow jar"() {
+        def shadowJar = file("build/libs/hello-world-1.0-all.jar")
+
+        given:
+        settingsFile << "rootProject.name = 'hello-world'"
+        buildFile << """
+            plugins {
+                id "io.micronaut.minimal.application"
+                id "com.github.johnrengelman.shadow" version "7.1.0"
```

Comment:
we currently generate apps using 7.1.2 of shade.

Reply:
Right, it doesn't really matter, it's a test, but I'll upgrade.